lib/blob.js

```diff
@@ -147,17 +240,164 @@ function Blob(options) {
+ *                                    // It must be omitted if it has been specified in the associated stored access policy.
+ *     read:              false,      // Read the content, properties, metadata or block list of a blob or of
+ *                                    // any blob in the container if the resourceType is a container.
+ *     add:               false,      // Add a block to an append blob or to any append blob if the resourceType is a container.
+ *     create:            false,      // Write a new blob, snapshot a blob, or copy a blob to a new blob. These operations can be done to any blob in the container
+ *                                    // if the resourceType is a container.
+ *     write:             false,      // Create or write content, properties, metadata, or block list. Snapshot or lease the blob. Resize the blob (page blob only).
+ *                                    // These operations can be done for every blob in the container if the resourceType is a container
+ *     delete:            false,      // Delete the blob or any blob in the container if the resourceType is a container.
+ *     list:              false,      // List blobs in the container.
+ *   },
+ *   cacheControl:        '...',      // The value of the Cache-Control response header to be returned. (optional)
+ *   contentDisposition:  '...',      // The value of the Content-Disposition response header to be returned. (optional)
+ *   contentEncoding:     '...',      // The value of the Content-Encoding response header to be returned. (optional)
+ *   contentLanguage:     '...',      // The value of the Content-Language response header to be returned. (optional)
+ *   contentType:         '...',      // The value of the Content-Type response header to be returned. (optional)
+ *   ipAddresses:         '...',      // An IP address or a range of IP addresses. (optional)
+ *   accessPolicy:        '...'       // Reference to stored access policy (optional)
+ *                                    // A GUID string
+ * }
+ * ```
+ * @returns {string} Shared-Access-Signature on string form.
+ *
+ */
+Blob.prototype.sas = function sas(container, blob, options){
+  // verify the required options
+  assert(options, "options is required");
+  assert(options.resourceType, 'options.resourceType is required');
+  assert(options.resourceType === 'b' || options.resourceType === 'c',
+    'The possible values for options.resourceType are `b` or `c`');
+  assert(options.permissions || options.accessPolicy, "options.permissions or options.accessPolicy must be specified");
+
+  // Check that we have credentials
+  if (!this.options.accountId ||
+    !this.options.accessKey) {
+    throw new Error("accountId and accessKey are required for SAS creation!");
+  }
+
+  // Construct query-string with required parameters
+  var query = {
+    sv:   SERVICE_VERSION,
+    sr:   options.resourceType,
+    spr:  'https'
+  }
+
+  if (options.expiry) {
```

Comment:
Technically, I think expiration is optional... But I agree.. if people don't set expiration or policy identifer, then there is no way to revoke the SAS, other than rotating the storage credentials.

I'm open to supporting it, but we should probably have people set an extra flag so that they've clearly communicated that they intended to do something sketchy...
Maybe `options.allowIrrevocableSAS = true`, or something like that, maybe a better name is possible..

Then we would have:
```js
if (!options.expiry && !options.accessPolicy && !options.allowIrrevocableSAS) {
  throw new Error('Explanation that you are about to do something dangerous, if you really want to do that you must set: allowIrrevocableSAS = true');
} 
``` 

Just an idea... IMO dangerous things are okay, as long as people confirm their intent to shoot themselves in the foot :)

Note: this could be a separate PR, that does it consistently across queue/table/blob